src/dbally/views/sqlalchemy_base.py

```diff
@@ -64,6 +74,16 @@ async def _build_filter_bool_op(self, bool_op: syntax.BoolOp) -> sqlalchemy.Colu
             return alchemy_op(await self._build_filter_node(bool_op.child))
         raise ValueError(f"BoolOp {bool_op} has no children")
 
+    async def apply_aggregation(self, aggregation: IQLQuery) -> None:
+        """
+        Creates a subquery based on existing
+
+        Args:
+            aggregation: IQLQuery object representing the filters to apply
+        """
+        self._subquery = self.get_subquery()
+        self._subquery = await self.call_aggregation_method(aggregation.root)
```

Comment:
This seems very confusing. You assign a value to `self._subquery` only to overwrite it on the very next line be another value that is **not** a subquery object (just a normal `Select`).

I don't see why the class needs a `_subquery` property at all. I would remove `self._subquery` and replace the entire method with this line:

```
self._select = await self.call_aggregation_method(aggregation.root)
```

This assumes that the `call_aggregation_method` is responsible for creating the subquery obejct (which seems reasonable to me). Alternatively you can modify `call_aggregation_method` to take the subquery object as an argument.

This will also allow you to revert all the changes to the `execute` method.